pkg/controllers/routercerts/controller.go

```diff
@@ -100,6 +100,11 @@ func (c *routerCertsDomainValidationController) sync(ctx context.Context, syncCt
 	// set the condition anywhere in sync() to update the controller's degraded condition
 	var condition operatorv1.OperatorCondition
 	defer func() {
+		if len(condition.Type) == 0 {
```

Comment:
how about adding a helper function for checking if a condition is empty ?

Reply:
to save a length check?  No.